src/Mvc/Mvc.Core/src/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -138,5 +190,50 @@ private static void InferContentTypes(ActionContext context, ObjectResult result
                 result.ContentTypes.Add("application/problem+xml");
             }
         }
+
+        private Task<object> EnumerateAsyncEnumerable(IAsyncEnumerable<object> value)
+        {
+            var type = value.GetType();
+            if (!_asyncEnumerableConverters.TryGetValue(type, out var result))
+            {
+                var enumerableType = ClosedGenericMatcher.ExtractGenericInterface(type, typeof(IAsyncEnumerable<>));
+                result = null;
+                if (enumerableType != null)
```

Comment:
Should this ever fail? This seems like you should just let it blow up right?

Reply:
Yup. Left over from before. I'll Debug.Assert it.